### PR TITLE
Replace participated with voting_for in Account.t

### PR DESCRIPTION
**Pull request by ghost-not-in-the-shell**
This PR did 2 things:
1) replaced the `participated:bool` field with `voting_for:state_hash` in `Account.t`;
2) created a lib for smallest out of scope field element

### CODE REVIEW

File: src/app/cli/src/graphql.ml [hunk 1]
```diff
@@ -204,11 +204,12 @@ module Make (Commands : Coda_commands.Intf) = struct
                 ~args:Arg.[]
                 ~resolve:(fun _ account ->
                   Stringable.public_key account.Account.Poly.delegate )
-            ; field "participated" ~typ:(non_null bool)
+            ; field "votingFor" ~typ:(non_null string)
                 ~doc:"TODO, not sure what this is"
```
Comment (imeckler) [inline src/app/cli/src/graphql.ml]: Can you add a good doc here?

File: src/lib/coda_base/account.ml [hunk 2]
```diff
@@ -163,58 +168,56 @@ let typ : (var, value) Typ.t =
...
-  ; participated= Boolean.var_of_value participated }
+  ; voting_for= State_hash.var_of_t voting_for }
 
 let var_to_triples
-    Poly.
-      {public_key; balance; nonce; receipt_chain_hash; delegate; participated}
+    Poly.{public_key; balance; nonce; receipt_chain_hash; delegate; voting_for}
     =
-  let%map public_key = Public_key.Compressed.var_to_triples public_key
+  let%bind public_key = Public_key.Compressed.var_to_triples public_key
```
Comment (imeckler) [inline src/lib/coda_base/account.ml]: It doesn't matter but just fyi you can include the `voting_for` in the `%map` using `and`.

File: src/lib/null_field_element/gen_values/dune [hunk 3]
```diff
@@ -0,0 +1,10 @@
+(executable
+  (name gen_values)
+  (libraries async core snark_params
+             ppxlib ppx_util)
+  (preprocessor_deps ../../../config.mlh)
+  (preprocess
+    (pps ppx_jane ppxlib.metaquot))
+  (flags -w -40 -g -warn-error
```
Comment (imeckler) [inline src/lib/null_field_element/gen_values/dune]: Do you need all these warnings turned off?

File: src/lib/null_field_element/gen_values/dune [hunk 4]
```diff
@@ -0,0 +1,10 @@
+(executable
+  (name gen_values)
+  (libraries async core snark_params
+             ppxlib ppx_util)
+  (preprocessor_deps ../../../config.mlh)
```
Comment (imeckler) [inline src/lib/null_field_element/gen_values/dune]: Do you need config.mlh?

File: src/lib/null_field_element/dune [hunk 5]
```diff
@@ -0,0 +1,14 @@
+(library
+  (name null_field_element)
```
Comment (imeckler) [inline src/lib/null_field_element/dune]: I think we should have a somewhat more descriptive name. Maybe like `non_curve_x_coordinate` or `outside_pedersen_image` or something?

**Comment by jkrauska** [comment]
@ghost-not-in-the-shell  can you resolve the lite_compat conflict?

**Pull request merged**